### PR TITLE
Use transaction in V102 migration

### PR DESCRIPTION
**Pull request by zeripath**
The code for `dropTableColumns` has a slightly confusing portion whereby the session is committed for MSSQL but not for other variants.

The v102 migration doesn't actually start a transaction so this weirdness does not affect it. However it probably should attempt to run this in a transaction.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -464,8 +464,6 @@ func dropTableColumns(sess *xorm.Session, tableName string, columnNames ...strin
 			sess.Rollback()
 			return fmt.Errorf("Drop table `%s` columns %v: %v", tableName, columnNames, err)
 		}
-
-		return sess.Commit()
```
Comment (silverwind) [inline models/migrations/migrations.go:469]: Is it safe to no longer commit here? It would affect all migrations, right?

Comment (zeripath) [reply models/migrations/migrations.go:469]: I have checked. This commit only occurs on the MSSQL branch and is the reason why this bug has occurred.

This is the problem with that sess.Commit() it gives the wrong impression.

Users of dropTableColumns have to commit their transactions at the end of the function.

**Review by silverwind — APPROVED** [review]

**Comment by zeripath** [comment]
I've re read this and I see that v102 didn't actually start a transaction so doesn't actually suffer loss - however it really should be in a transaction.

**Comment by lafriks** [comment]
In such case rollback should also be moved out of that function

File: models/migrations/migrations.go [hunk 2]
```diff
@@ -464,8 +464,6 @@ func dropTableColumns(sess *xorm.Session, tableName string, columnNames ...strin
 			sess.Rollback()
 			return fmt.Errorf("Drop table `%s` columns %v: %v", tableName, columnNames, err)
 		}
```
Comment (lunny) [inline models/migrations/migrations.go:465]: Also removed all `sess.Rollback() from the function?

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@zeripath Why removed the backport label?

**Comment by zeripath** [comment]
Because although I initially thought that this would always cause data loss I put the backport label in - now I understand that it doesn't start a transaction, the data-loss risk is much much lower and so it probably doesn't need backporting.

File: models/migrations/migrations.go (hunk 2 above)
Comment (zeripath) [reply models/migrations/migrations.go:465]: done

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
conflicted.

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**